src/Logger.js

```diff
@@ -0,0 +1,66 @@
+import {Map, List} from 'immutable';
+
+const logLevels = List([
+    {
+        name: 'error',
+        consoleMethod: 'error'
+    },
+    {
+        name: 'warn',
+        consoleMethod: 'warn'
+    },
+    {
+        name: 'info',
+        consoleMethod: 'log'
+    },
+    {
+        name: 'verbose',
+        consoleMethod: 'log'
+    },
+    {
+        name: 'silly',
+        consoleMethod: 'log'
+    }
+]);
+
+var Logger = {
+
+    logLevel: 0,
+
+    setLogLevel: function(level) {
+        this.logLevel = this.getLevelIndex(level);
+
+        // create methods on Logger for each logLevel
+        logLevels.map(({name, consoleMethod}, key) => {
```

Comment:
Should probably be a forEach seeing as there is no return and there are side effects.